src/Display/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import Modal from '.';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { fireEvent, render } from '@testing-library/react';
+
+const props = {
+  title: 'Modal Title',
+  content: 'Modal Content',
+  footer: 'Modal Footer',
+  onClose: jest.fn(),
+};
+
+const OpenedModal = (
+  <Modal
+    title={props.title}
+    isVisible={true}
+    footer={[<button key="button">{props.footer}</button>]}
+    onClose={props.onClose}
+  >
+    <p>{props.content}</p>
+  </Modal>
+);
+
+function setupModal(isVisible: boolean) {
+  const onClose = jest.fn();
+  const ModalComponent = (
+    <Modal
+      title={props.title}
+      isVisible={isVisible}
+      footer={[<button key="button">{props.footer}</button>]}
+      onClose={onClose}
+    >
+      <p>{props.content}</p>
+    </Modal>
+  );
+  const { getByTestId, getByRole } = render(ModalComponent);
+  const modalContainer = getByTestId('modal-container');
+  const modalDialog = getByRole('dialog');
+  return [modalContainer, onClose, modalDialog];
+}
+
+it('<Modal> should render with a title, content, footer and an onClick handler', () => {
+  const ModalSnapshot = renderer.create(OpenedModal).toJSON();
+  expect(ModalSnapshot).toMatchSnapshot();
+});
+
+describe('when modal is opened', () => {
+  it('should be visible', () => {
+    const [modalContainer] = setupModal(true);
+    expect(modalContainer).toHaveStyle('visibility: visible');
+  });
+
+  it('should not be closed if the modal is clicked', () => {
+    const [, onClose, modalDialog] = setupModal(true);
+    fireEvent.click(modalDialog as Element);
+    expect(onClose).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe('when modal is opened', () => {
+  it('should display the title', () => {
+    const { getByText } = render(OpenedModal);
+    const modalHeader = getByText(props.title);
+    expect(modalHeader).toHaveTextContent(props.title);
+  });
+
+  it('should display the content', () => {
+    const { getByText } = render(OpenedModal);
+    const modalContent = getByText(props.content);
+    expect(modalContent).toHaveTextContent(props.content);
+  });
+
+  it('should display the footer', () => {
+    const { getByText } = render(OpenedModal);
+    const modalFooter = getByText(props.footer);
+    expect(modalFooter).toHaveTextContent(props.footer);
```

Comment:
hey @franciscrispin , this checks that we have a `textContent`. 

In `setupModal`, the modal is given a `button` element that will then render `props.footer`.

I believe that we should also check that the button is rendered.
And that the button has this text content?

When I bring in the snapshot test, then it starts to get clearer also.
I understand that snapshot testing would also inspect HTML elements and their children?

Reply:
Okay will check for the button!

Yeap, snapshot testing would also inspect HTML elements and their children.